onnxruntime/core/graph/graph.cc

```diff
@@ -572,23 +573,56 @@ const Path& Node::ModelPath() const noexcept {
 
 #if !defined(ORT_MINIMAL_BUILD)
 
-Function* Node::GetMutableFunctionBody(bool try_init_func_body) {
-  if (nullptr != func_body_) {
-    return func_body_;
-  }
+bool Node::CanBeInlined() const {
+  return func_body_ || func_template_ || op_ && (op_->HasFunction() || op_->HasContextDependentFunction());
+}
 
+std::unique_ptr<Function> Node::GetInstantiateFunctionBody(const logging::Logger& logger) const {
```

Comment:
Nit: `DoInstantiateFunctionBody` or `InstantiateFunctionBodyImpl` may be better names than `GetInstantiateFunctionBody`. Alternatively, we can just merge this into `InstantiateFunctionBody` ... I haven't checked if this is called anywhere else.

Reply:
in the gradient_op tester, there is a code that need to check the provider ability (including whether it has kernel for the function body) before executing the model. so I need the ability to get function body without touch the graph.